src/include/duckdb/common/types/column_data_collection.hpp

```diff
@@ -135,9 +135,9 @@ class ColumnDataCollection {
 	void ScanAtIndex(ColumnDataParallelScanState &state, ColumnDataLocalScanState &lstate, DataChunk &result,
 	                 idx_t chunk_index, idx_t segment_index, idx_t row_index) const;
 
-private:
 	//! Initialize the column data collection
 	void Initialize(vector<LogicalType> types);
+private:
```

Comment:
@Mytherin I had to expose the `Initialize` method to get the test working. Let me know if you prefer me to keep it private and initialize it some other way (I would have to expose the `copy_functions` collection though)